src/middleware/ChaosHandler.ts

```diff
@@ -0,0 +1,289 @@
+	 * @param {ChaosHandlerOptions} [options = new ChaosHandlerOptions()] - The testing handler options instance
+	 * @param manualMap - The Map passed by user containing url-statusCode info
+	 * @returns An instance of Testing Handler
+	 */
+	public constructor(options: ChaosHandlerOptions = new ChaosHandlerOptions(), manualMap?: Map<string, Map<string, number>>) {
+		this.options = options;
+		this.manualMap = manualMap;
+	}
+
+	/**
+	 * Generates responseHeader
+	 * @private
+	 * @param {number} statusCode - the status code to be returned for the request
+	 * @param {string} requestID - request id
+	 * @param {string} requestDate - date of the request
+	 * @returns response Header
+	 */
+	private createResponseHeaders(statusCode: number, requestID: string, requestDate: string) {
+		const responseHeader: Headers = new Headers();
+
+		responseHeader.append("Cache-Control", "no-store");
+		responseHeader.append("request-id", requestID);
+		responseHeader.append("client-request-id", requestID);
+		responseHeader.append("x-ms-ags-diagnostic", "");
+		responseHeader.append("Date", requestDate);
+		responseHeader.append("Strict-Transport-Security", "");
+
+		if (statusCode === 429) {
+			// throttling case has to have a timeout scenario
+			responseHeader.append("retry-after", "300");
+		}
+		return responseHeader;
+	}
+
+	/**
+	 * Generates responseBody
+	 * @private
+	 * @param {number} statusCode - the status code to be returned for the request
+	 * @param {string} statusMessage - the status message to be returned for the request
+	 * @param {string} requestID - request id
+	 * @param {string} requestDate - date of the request
+	 * @returns response body
+	 */
+	private createResponseBody(statusCode: number, statusMessage: string, requestID: string, requestDate: string) {
+		let responseBody: any;
+		if (statusCode >= 400) {
+			const codeMessage: string = httpStatusCode[statusCode];
+			const errMessage: string = statusMessage;
+
+			responseBody = {
+				error: {
+					code: codeMessage,
+					message: errMessage,
+					innerError: {
+						"request-id": requestID,
+						date: requestDate,
+					},
+				},
+			};
+		} else {
+			responseBody = {};
+		}
+		return responseBody;
+	}
+
+	/**
+	 * creates a response or passes the request to httpMessageHandler
+	 * @private
+	 * @param {ChaosHandlerOptions} ChaosHandlerOptions - The ChaosHandlerOptions object
+	 * @param {string} requestURL - the URL for the request
+	 * @param {RequestMethod} requestMethod - enum for request method
+	 * @returns nothing
+	 */
+	private async createResponse(chaosHandlerOptions: ChaosHandlerOptions, context: Context): Promise<void> {
+		try {
+			let responseBody: any;
+			let responseHeader: Headers;
+			let requestID: string;
+			let requestDate: Date;
+			const requestURL = context.request as string;
+			const requestMethod = context.options.method as RequestMethod;
+
+			const passThrough = this.setStatusCode(chaosHandlerOptions, requestURL, requestMethod);
```

Comment:
it doesn't make sense to return passthrough value from setStatusCode method. better have the pass through logic in a separate function.